app/components/language-option.vue

```diff
@@ -0,0 +1,41 @@
+<template>
+  <span @click="setLanguage">
+    <img 
+      :src="flagIcon" 
+      alt="flag icon" 
+      class="flag-icon" />
+  </span>
+</template>
+
+<script>
+  import { mapState, mapMutations } from 'vuex'
+
+  export default {
+    name: 'language-option',
+
+    props: {
+      language: Object
+    },
+
+    computed: {
+      flagIcon () {
+        return this.language.image
+      }
+    },
+
+    methods: {
+      setLanguage () {
+        this.$store.commit('SET_LANGUAGE', this.language.value)
```

Comment:
Yeah it's just a preference thing and to avoid importing it in every component, but we can go with the import if you like that better! :)